lib/json_api_controller/updatable_resource.rb

```diff
@@ -11,38 +11,42 @@ module UpdatableResource
     end
 
     def update
-      ActiveRecord::Base.transaction do
-        build_resource_for_update(update_params)
-        controlled_resource.save!
+      resource_class.transaction do
```

Comment:
Should we protect against  side effects in nested transactions like create? `resource_class.transaction(requires_new: true)`